### ISSUE TITLE
error| syntax error near unexpected token `(' #1799

### ISSUE THREAD

**Issue by corbolais** [itspriddle/vim-shellcheck#9]
Originally created [shellcheck](https://github.com/koalaman/shellcheck/) issue: https://github.com/koalaman/shellcheck/issues/1799 .

* My shellcheck version (shellcheck --version or "online"): 0.7.0
* debian package version: 0.7.0-2
* tried on shellcheck.net: OK, no error
* tried with `vim-shellcheck`, as of git SHA 4346419ac57ef341a15aa39c827c0848f17c6faf: FAIL

Including report for direct reference:

* Here's a snippet that shows the problem:

#!/bin/bash

```sh
shopt -s extglob;
for f in /etc/hosts/host\.!(mail)\.conf ; do
    echo "$f";
done;
```

* Here's what shellcheck currently says:
error| syntax error near unexpected token `('

* Expected: No error

**Comment by itspriddle** [comment]
@corbolais thanks for the report. Can you provide your Vim version? And are you using any other bash/shell related plugins? 

I can't reproduce the error with your script on either macOS (Vim 8.2.250) or Ubuntu (Vim 8.0.1453) when using Shellcheck 0.7.0.

**Comment by corbolais** [comment]
Using debian vim package version 2:8.1.2269-1 i386 .

VIM plugins installed:

- .vim/bundle/Apprentice  commit 0e624fd7efdc96d3fa3c1e12f7fc701fa1534eaf
- .vim/bundle/powerline  commit d498f29261afa0799aaa95924744dbf55b6960d5
- .vim/bundle/syntastic  commit 39b35b23b952d620b8ec7cabb13110f586663837
- .vim/bundle/vim-shellcheck  commit 4346419ac57ef341a15aa39c827c0848f17c6faf

**Comment by bewuethr** [comment]
My guess is that your error is caused by syntastic; the `sh` linter uses `-n` under the hood, which results in your error, see [here](https://github.com/vim-syntastic/syntastic/blob/f3766538720116f099a8b1517f76ae2f094afd20/syntax_checkers/sh/sh.vim#L37-L46).

**Comment by corbolais** [comment]
Good point, looks suspicious at a glance.

Gosh, just went to blame [`vim-shellcheck`](https://github.com/itspriddle/vim-shellcheck/issues/9).

Thank you.
